dbms/src/Storages/KVStore/ProxyStateMachine.h

```diff
@@ -0,0 +1,421 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Common/Logger.h>
+#include <Core/TiFlashDisaggregatedMode.h>
+#include <Interpreters/Settings.h>
+#include <Poco/Util/AbstractConfiguration.h>
+#include <Poco/Util/LayeredConfiguration.h>
+#include <Server/ServerInfo.h>
+#include <Storages/FormatVersion.h>
+#include <Storages/KVStore/FFI/ProxyFFI.h>
+#include <Storages/KVStore/KVStore.h>
+#include <Storages/KVStore/TMTContext.h>
+
+#include <boost/noncopyable.hpp>
+#include <chrono>
+#include <string>
+#include <thread>
+#include <unordered_map>
+#include <vector>
+
+namespace DB
+{
+
+extern "C" {
+void run_raftstore_proxy_ffi(int argc, const char * const * argv, const EngineStoreServerHelper *);
+}
+
+/// Manages the argument being passed to proxy, through `run_raftstore_proxy_ffi` call.
+// It is different from `TiFlashRaftConfig` which serves computing.
```

Comment:
> // It is different from `TiFlashRaftConfig` which serves computing.

I don't know what this line means. What does `it` means here?

Reply:
"It" stands for `TiFlashProxyConfig`.